src/block/DragonEgg.php

```diff
@@ -57,7 +58,9 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 	}
 
 	public function onAttack(Item $item, int $face, ?Player $player = null) : bool{
-		$this->teleport();
+		if($player !== null && !$player->getGamemode()->equals(GameMode::CREATIVE())){
+			$this->teleport();
+		}
 		return true;
```

Comment:
This should return false if nothing was done, otherwise it may have side effects.